src/test_c2s_srv.c

```diff
@@ -388,15 +407,20 @@ int test_c2s(int ctlsockfd, tcp_stat_agent* agent, TestOptions* testOptions,
              testOptions->child0);
     log_println(1, "%s", buff);
     snprintf(buff, sizeof(buff), "%0.0f", *c2sspd);
-    send_json_message(ctlsockfd, TEST_MSG, buff, strlen(buff), testOptions->connection_flags, JSON_SINGLE_VALUE);
+    send_json_message_any(ctl, TEST_MSG, buff, strlen(buff),
+                          testOptions->connection_flags, JSON_SINGLE_VALUE);
 
     // get receiver side Web100 stats and write them to the log file. close
     // sockets
     if (record_reverse == 1)
-      tcp_stat_get_data_recv(recvsfd, agent, conn, count_vars);
+      tcp_stat_get_data_recv(c2s_conn.socket, agent, conn, count_vars);
 
 
-    close(recvsfd);
+    if (c2s_conn.ssl != NULL) {
+      SSL_shutdown(c2s_conn.ssl);
+      SSL_free(c2s_conn.ssl);
```

Comment:
Good practice to set it to `NULL` after freeing to limit the possibility of double-frees